lib/iopromise/deferred/executor_pool.rb

```diff
@@ -17,19 +17,28 @@ def execute_continue
           next_batch
         end
 
-        until @current_batch.empty?
```

Comment:
This loop used to be needed a while ago, but in the more recent revisions of the main event loop, this function is correctly called each loop iteration and it's safe for it not to progress/complete fully. I've removed the loop here, since it's simpler and better to rely on the main loop.